chromadb/api/async_fastapi.py

```diff
@@ -0,0 +1,634 @@
+            embedding_function=embedding_function,
+            data_loader=data_loader,
+            get_or_create=True,
+            tenant=tenant,
+            database=database,
+        )
+
+    @trace_method("AsyncFastAPI._modify", OpenTelemetryGranularity.OPERATION)
+    @override
+    async def _modify(
+        self,
+        id: UUID,
+        new_name: Optional[str] = None,
+        new_metadata: Optional[CollectionMetadata] = None,
+    ) -> None:
+        await self._make_request(
+            "put",
+            "/collections/" + str(id),
+            json={"new_metadata": new_metadata, "new_name": new_name},
+        )
+
+    @trace_method("AsyncFastAPI.delete_collection", OpenTelemetryGranularity.OPERATION)
+    @override
+    async def delete_collection(
+        self,
+        name: str,
+        tenant: str = DEFAULT_TENANT,
+        database: str = DEFAULT_DATABASE,
+    ) -> None:
+        await self._make_request(
+            "delete",
+            "/collections/" + name,
+            params={"tenant": tenant, "database": database},
+        )
+
+    @trace_method("AsyncFastAPI._count", OpenTelemetryGranularity.OPERATION)
+    @override
+    async def _count(
+        self,
+        collection_id: UUID,
+    ) -> int:
+        """Returns the number of embeddings in the database"""
+        resp_json = await self._make_request(
+            "get",
+            "/collections/" + str(collection_id) + "/count",
+        )
+
+        return cast(int, resp_json)
+
+    @trace_method("AsyncFastAPI._peek", OpenTelemetryGranularity.OPERATION)
+    @override
+    async def _peek(
+        self,
+        collection_id: UUID,
+        n: int = 10,
+    ) -> GetResult:
+        return await self._get(
+            collection_id,
+            limit=n,
+            include=["embeddings", "documents", "metadatas"],
+        )
+
+    @trace_method("AsyncFastAPI._get", OpenTelemetryGranularity.OPERATION)
+    @override
+    async def _get(
+        self,
+        collection_id: UUID,
+        ids: Optional[IDs] = None,
+        where: Optional[Where] = {},
+        sort: Optional[str] = None,
+        limit: Optional[int] = None,
+        offset: Optional[int] = None,
+        page: Optional[int] = None,
+        page_size: Optional[int] = None,
+        where_document: Optional[WhereDocument] = {},
+        include: Include = ["metadatas", "documents"],
+    ) -> GetResult:
+        if page and page_size:
+            offset = (page - 1) * page_size
+            limit = page_size
+
+        resp_json = await self._make_request(
+            "post",
+            "/collections/" + str(collection_id) + "/get",
+            json={
+                "ids": ids,
+                "where": where,
+                "sort": sort,
+                "limit": limit,
+                "offset": offset,
+                "where_document": where_document,
+                "include": include,
+            },
+        )
+
+        return GetResult(
```

Comment:
could we code share the logic after the resp is back?

Reply:
I did this for Collection, but I think sharing code between the HTTP client implementations will be much easier after both async & sync are using httpx in a future PR. Then the shared logic can take a common set of arguments, return a `httpx.Request`, receive a `httpx.Response` back, and then do whatever transforms are necessary.

Reply:
ok sounds good lets clean this up then